src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs

```diff
@@ -51,5 +51,35 @@ public ActionResult UploadPrenotifications(Guid notificationId, PrenotificationB
 
             return View(model);
         }
+
+        [HttpGet]
+        public ActionResult Warning(Guid notificationId)
+        {
+            var model = new WarningChoiceViewModel(notificationId);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Warning(WarningChoiceViewModel model, string cfp)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            if (model.GetEnumDisplayValue(WarningChoicesList.Leave).Equals(model.WarningChoices.SelectedValue))
+            {
+                return RedirectToAction("Index", "Options", new { area = "NotificationApplication", id = model.NotificationId });
+            }
+            else if (model.GetEnumDisplayValue(WarningChoicesList.Return).Equals(model.WarningChoices.SelectedValue))
+            {
+                // To do: Send user to upload page for the shipment movement document
+                throw new NotImplementedException("Redirection to upload page for the shipment movement document not yet implemented");
+            }
+
+            throw new InvalidOperationException("Radio selection not per expectations");
```

Comment:
I'd say just return the back to the same `View` for this line.

Reply:
Change made